### PR TITLE
Implement upgrade to study version 9.0

### PR DESCRIPTION
**Pull request by flomnes**
- Add empty study template
- Add tests
- Fix small typo with DOTTED_VERSION

## This works, but...
Unlike other tests, we don't run `UpgradeTo0900` directly because it does nothing. We run the app directly, since it has the job of updating file `study.antares`.

This won't work for e.g version 9.1 since we'll run `UpgradeTo0900` then `UpgradeTo0901`, but `study.antares` will stay untouched.

There's probably a design problem.

### CODE REVIEW

File: tests/upgrade_app/test_upgrade_0900.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
+# from antares.study.version.upgrade_app.upgrader_0900 import UpgradeTo0900
+from antares.study.version.upgrade_app import UpgradeApp
+from antares.study.version import StudyVersion
+from antares.study.version.ini_reader import IniReader
+from tests.conftest import StudyAssets
+
+
+def get_version(d):  # type: ignore
```
Comment (MartinBelthle) [inline tests/upgrade_app/test_upgrade_0900.py:8]: Not really sure you need to introduce a method to do this

File: tests/upgrade_app/test_upgrade_0900.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+    """
+    Extract version attribute from a nested dict structure
+    """
+    return d["antares"]["version"]
+
+
+def test_nominal_case(study_assets: StudyAssets):
```
Comment (MartinBelthle) [inline tests/upgrade_app/test_upgrade_0900.py:15]: I think you should also check that studies are the same (cf. other tests):
```python
assert are_same_dir(study_assets.study_dir, study_assets.expected_dir)
```

File: tests/upgrade_app/test_upgrade_0900.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+# from antares.study.version.upgrade_app.upgrader_0900 import UpgradeTo0900
```
Comment (MartinBelthle) [inline tests/upgrade_app/test_upgrade_0900.py:1]: You can remove this unused import

File: tests/create_app/test_create_app.py [hunk 4]
```diff
@@ -40,6 +40,6 @@ def test_create_app(self, tmp_path: Path, study_version: StudyVersion):
             "version": expected_version,
         }
 
-        created_date = datetime.datetime.fromtimestamp(properties["created"])
-        last_save_date = datetime.datetime.fromtimestamp(properties["lastsave"])
+        created_date = datetime.datetime.fromtimestamp(int(properties["created"]))
```
Comment (MartinBelthle) [inline tests/create_app/test_create_app.py:43]: You had to change this because of ConfigParser ?

**Review by MartinBelthle — COMMENTED** [review]
We can tackle the design issue inside another PR, so don't bother

File: tests/upgrade_app/test_upgrade_0900.py (hunk 2 above)
Comment (flomnes) [reply tests/upgrade_app/test_upgrade_0900.py:15]: Done, ignoring study.antares

Main differences :
- dates are decimal
- properties are out of order

File: tests/upgrade_app/test_upgrade_0900.py (hunk 1 above)
Comment (flomnes) [reply tests/upgrade_app/test_upgrade_0900.py:8]: I wanted to create a lambda function but mypy doesn't like them

**Review by MartinBelthle — APPROVED** [review]

**Pull request merged**